File: 0647-palindromic-substrings/0647-palindromic-substrings.java
class Solution {

    public boolean check(String s) {
        String temp = s;

        String rev = new StringBuilder(s).reverse().toString();

        return rev.equals(temp);
    }

    public int countSubstrings(String s) {
        int n = s.length();
        int count = 0;

        for (int i = 0; i < n; i++) {
            String temp = "";

            for (int j = i; j < n; j++) {
                temp = temp + s.charAt(j);

                if (check(temp))
                    count++;
            }
        }

        return count;
    }
}

File: 0647-palindromic-substrings/0647-palindromic-substrings.cpp
class Solution {
public:
    bool check(string s) {
        string temp = s;
        reverse(temp.begin(), temp.end());

        if (temp == s)
            return true;

        return false;
    }
    int countSubstrings(string s) {
        int n = s.size();

        int count = 0;

        for (int i = 0; i < n; i++) {
            string temp = "";
            for (int j = i; j < n; j++) {
                temp = temp + s[j];
                // cout << temp << " ";
                if (check(temp))
                    count++;
            }
        }

        return count;
    }
};